proxy/http/HttpSM.h

```diff
@@ -569,9 +545,7 @@ class HttpSM : public Continuation, public PluginUserArgs<TS_USER_ARGS_TXN>
 
   HttpVCTable vc_table;
 
-  IOBufferReader *ua_raw_buffer_reader = nullptr;
-
-  HttpVCTableEntry *ua_entry     = nullptr;
+  HttpUserAgent _ua{};
```

Comment:
It looks like we could do a similar thing for `server_entry` and `server_txn`. What would it be called? `HttpServer`? It would be nice if the two had a same prefix or suffix in their names to indicate they are basically the same thing but one for client side and one for server side.

Reply:
`HttpServer` seems like a good name. It might be a good idea to try this in a separate PR, unless you are ok with having a pretty large diff? This one is already a little bigger than I thought would be nice to review.

Reply:
I hadn't thought as far ahead as extracting `HttpServer` and making a base class, but I think that's a good direction to take this.